src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -207,25 +196,9 @@ const FixPhoneNumbers: React.FC<Props> = ({
     handleDeleteModalClose();
   };
 
-  const handleAdd = (personId: string, number: string): void => {
+  const handleChange = (personId: string, numbers: PhoneNumberData[]): void => {
     const temp = { ...dataState };
-    temp[personId].phoneNumbers.push({
-      updatedAt: new Date().toISOString(),
-      number: number,
-      primary: false,
-      source: 'MPDX',
-    });
-    setDataState(temp);
-  };
-
-  const handleChangePrimary = (personId: string, numberIndex: number): void => {
-    const temp = { ...dataState };
-    temp[personId].phoneNumbers = temp[personId].phoneNumbers.map(
-      (number, index) => ({
-        ...number,
-        primary: index === numberIndex ? true : false,
-      }),
-    );
+    dataState[personId].phoneNumbers = numbers;
```

Comment:
It is hard to tell how much you changed the delete function in this PR since it looks like you removed it and then added it back in. I did notice that if you delete all of the phone numbers it throws an error because it tries to set an unknown phone number to primary.